File: Selenium/Activities/TestActivity11.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

# Start the Driver
with webdriver.Firefox() as driver:
    # Declare the actions variable
    wait = WebDriverWait(driver,timeout=10)
    # Navigate to the URL
    driver.get("https://training-support.net/webelements/dynamic-controls")
    # Print the title of the page
    print("Page title is: ", driver.title)

    checkbox = driver.find_element(By.ID,"checkbox")
    print("Checkbox is visible? ", checkbox.is_displayed())
    toggle = driver.find_element(By.XPATH,"//button[text()='Toggle Checkbox']")
    
    toggle.click()
    wait.until(EC.invisibility_of_element(checkbox))
    print("Checkbox is visible? ", checkbox.is_displayed())

    toggle.click()
    wait.until(EC.element_to_be_clickable(checkbox))
    checkbox.click()

    print("is checbox selected? ", checkbox.is_selected())

driver.quit()

File: Selenium/Activities/TestActivity11.java
package activities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class TestActivity11 {

	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(5));
		
				
				driver.get("https://training-support.net/webelements/dynamic-controls");
		        // Print the title of the page
		        System.out.println("Page title: " + driver.getTitle());
		        
		        //Find the checkbox element
		        WebElement checkbox = driver.findElement(By.xpath("//input[@id='checkbox']"));
		        WebElement toggle = driver.findElement(By.xpath("//button[text()='Toggle Checkbox']"));
		        
		        //toggle the visibility
		        toggle.click();
		        
		        //wait for the checkbox to dissappear
		        wait.until(ExpectedConditions.invisibilityOf(checkbox));
		        
		        //toggle the visibility
		        toggle.click();
		        //wait for checkbox to appear and click it
		        wait.until(ExpectedConditions.visibilityOf(checkbox));
		        
		        driver.quit();

	}

}
